### PR TITLE
[Build] Update scripts to use dyld_info

### PR DESCRIPTION
**Pull request by drexin**
dyldinfo has been deprecated and will no longer be available on newer Xcode releases. It has been replaced with dyld_info. To stay compatible with older releases, we are still falling back to dyldinfo, if dyld_info is not available.

rdar://98570807

### CODE REVIEW

**Comment by drexin** [comment]
@swift-ci smoke test

File: utils/swift-darwin-postprocess.py [hunk 1]
```diff
@@ -31,9 +31,17 @@ def main(arguments):
...
-        ['xcrun', 'dyldinfo', '-dylibs', filename],
-        universal_newlines=True)
+    dylibsOutput = None
+    try:
+        # `dyldinfo` has been replaced with `dyld_info`, so we try it first
+        # before falling back to `dyldinfo`
+        dylibsOutput = subprocess.check_output(
+            ['xcrun', 'dyld_info', '-dependents', filename],
+            universal_newlines=True)
+    except exceptions:
```
Comment (glessard) [inline utils/swift-darwin-postprocess.py:41]:
```suggestion
    except:
```

File: utils/swift-rpathize.py [hunk 2]
```diff
@@ -54,9 +54,18 @@ def main(arguments):
...
-        universal_newlines=True)
+    dylibsOutput = None
+
+    try:
+        # `dyldinfo` has been replaced with `dyld_info`, so we try it first
+        # before falling back to `dyldinfo`
+        dylibsOutput = subprocess.check_output(
+            ['xcrun', 'dyld_info', '-dependents', filename],
+            universal_newlines=True)
+    except exceptions:
```
Comment (glessard) [inline utils/swift-rpathize.py:65]:
```suggestion
    except:
```

**Comment by drexin** [comment]
@swift-ci smoke test

File: utils/swift-darwin-postprocess.py [hunk 3]
```diff
@@ -31,9 +31,17 @@ def main(arguments):
...
-        ['xcrun', 'dyldinfo', '-dylibs', filename],
-        universal_newlines=True)
+    dylibsOutput = None
+    try:
+        # `dyldinfo` has been replaced with `dyld_info`, so we try it first
+        # before falling back to `dyldinfo`
+        dylibsOutput = subprocess.check_output(
+            ['xcrun', 'dyld_info', '-dependents', filename],
+            universal_newlines=True)
+    except:
```
Comment (drexin) [inline utils/swift-darwin-postprocess.py:41]:
```suggestion
    except subprocess.CalledProcessError:
```

File: utils/swift-rpathize.py [hunk 4]
```diff
@@ -54,9 +54,18 @@ def main(arguments):
...
-        universal_newlines=True)
+    dylibsOutput = None
+
+    try:
+        # `dyldinfo` has been replaced with `dyld_info`, so we try it first
+        # before falling back to `dyldinfo`
+        dylibsOutput = subprocess.check_output(
+            ['xcrun', 'dyld_info', '-dependents', filename],
+            universal_newlines=True)
+    except:
```
Comment (drexin) [inline utils/swift-rpathize.py:65]:
```suggestion
    except subprocess.CalledProcessError:
```

**Comment by drexin** [comment]
@swift-ci smoke test

**Comment by drexin** [comment]
@swift-ci smoke test macos

**Review by glessard — APPROVED** [review]
FWIW this fixed the problem on my machine!

**Pull request merged**